src/Marquee/Marquee.js

```diff
@@ -950,7 +950,7 @@ var MarqueeItem = {
 		var node = this.$.marqueeText ? this.$.marqueeText.hasNode() : this.hasNode(),
 			rect;
 
-		if (node && this._marquee_distance == null && this.getAbsoluteShowing()) {
+		if (node && this.getAbsoluteShowing()) {
```

Comment:
"measures every single time, even when it doesn't need to"

On the initial render alone, `_marquee_calcDistance` is called 7 times by every marquee. If you have a normal DataGridList, where each of the 21 items on screen has a caption and subCaption, each of which marquee, that's 294 calls for one screen update. `_marquee_calcDistance` preforms a costly DOM reading operation `getBoundingClientRect`, and the numbers add up quickly. We need to keep this function in particular as lean as possible with as much pre-caching as possible and exit quickly if possible.

`ItemOverlay` probably needs to get refactored to use the showing property instead of basic CSS hiding, so marquee can be aware of its state change.